00-open-a-shapefile.Rmd

```diff
@@ -78,237 +75,254 @@ Working With Shapefile Attributes in R </a>
+This metadata are associated with all types of shapefiles (point, line,
+polygon).  The specific attributes may vary but all shapefiles will have
+associated attributes.
 
-```
+We can view specific metadata for the shapefile:
 
+```{r view-metadata }
+#view just the class for the shapefile
+class(aoiBoundary_HARV)
 
-#View Shapefile Metadata
+#view just the crs for the shapefile
+crs(aoiBoundary_HARV)
 
-When we imported the `HarClip_UTMZ18` shapefile layer into R (as our `aoiBoundary` object) the `readOGR()` function automatically stored some key metadata including:
+#view just the extent for the shapefile
+extent(aoiBoundary_HARV)
 
-1. **Object Type:** - in this case our aoiBoundary object is a polygon of class `SpatialPolygonsDataFrame`
-2. **CRS** - projection information (`coord ref.` in the output above) [LINK TO SPATIAL DATA INTRO LESSON?]
-3. **Extent** - the spatial extent of the shapefile. Note that the spatial
-extent for a shapefile represents the extent for ALL polygons (or spatial objects)
-in the shapefile!
+#view all metadata at same time
+aoiBoundary_HARV
+```
 
-We can view the CRS, Extent and other metadata for the shapefile using the following
-commands:
+Our `aoiBoundary_HARV` object is a polygon of class `SpatialPolygonsDataFrame`, 
+in the projection UTM zone 18N. This information is critical to interpreting the
+object `extent`. 
 
-```{r view-crs-extent }
-#view just the crs for the shapefile
-crs(aoiBoundary)
+##Spatial Data Attributes
+In addition to the metadata, each object has one or more attributes linked 
+to it.  Attributes of an object can include measurements, descriptions, or
+classifications of the object. Attribute data can be any sort of information
+that you might want to associate with your spatial objects. Each spatial 
+object in a shapefile can have the same set of attributes or each can be 
+different. 
 
-#view just the extent for the shapefile
-extent(aoiBoundary)
+We can query all the attributes of the this `SpatialPolygonsDataFrame` by 
+using the `attributes()` function. 
 
+``` {r Shapefile-attributes-2}
+#alternate way to view attributes 
+attributes(aoiBoundary_HARV)
 ```
 
+## Metadata & Attribute Summary
+We can view a metadata & attribute summary of each shapefile by entering 
+the name of the `R` object in the console. Note that the metadata output
+includes the _class_, the number of _features_, the _extent_, and the coordinate
+reference system (`CRS`) of the `R` object. The last two lines of summary show a
+preview of the `R` object _attributes_.
 
-#Import a line and point shapefile
-
+```{r shapefile-summary}
+#view a summary of metadata & attributes associated with the spatial object
+summary(aoiBoundary_HARV)
 
->#Challenge: Import a line and point shapefile
->
->Using the steps above, import the HARV_roads and HARVtower_UTM18N layers into R.
->Call the Harv_roads object `lines` and the HARVtower_UTM18N `point`
->Answer the following questions:
+```
 
```

Comment:
Ok i see where you're going. How does it look? i'm thinking it could look a bit odd when we have lots of challenge output potentially.

Another option could be some sort of rule before and after the figures.

IN short - i think i'd need to see it but i'm inclined to just have the colored portions be the "instructions" for the challenge .
If you want, i'll rebuild and have a look to see?